SCClassLibrary/Common/Streams/Patterns.sc

```diff
@@ -642,17 +642,17 @@ Pprotect : FilterPattern {
 		^super.new(pattern).func_(func)
 	}
 	storeArgs { ^[ pattern, func ] }
-	asStream {
-		var rout = Routine(pattern.embedInStream(_));
-		rout.exceptionHandler = { |error|
-			// 'func' might throw an error
-			// we must clear the exceptionHandler before that
-			// otherwise, infinite recursion is the result
-			rout.exceptionHandler = nil;
-			func.value(error, rout);
-			nil.handleError(error)
-		};
-		^rout
+
+	embedInStream { arg inval;
```

Comment:
does this require a documentation change?

Reply:
No, IMO it does not require a documentation change. It's a bug fix -- I don't think the help has to say "this used to be broken but it isn't broken anymore."

Reply:
right, just asking because the method is now embedinstream rather than asstream

Reply:
Not a user facing change.

If a user calls asStream on a pattern that implements its behavior in embedInStream, it falls back to Pattern:asStream, which returns a Routine that calls embedInStream.

If a user calls embedInStream on a pattern that implements its behavior in asStream, then it does `^this.asStream.embedInStream(inval)` and falls back to Stream:embedInStream.

So a pattern's implementation can be in either method and it's completely transparent to the user. (Generally embedInStream is recommended.)

Reply:
cool, thanks for the explanation!